src/ray/raylet/worker_pool.h

```diff
@@ -111,20 +117,30 @@ class WorkerPool {
   int num_workers_per_process_;
 
  private:
+  /// An internal data structure that maintains the pool state per language.
+  struct State {
+    /// The commands and arguments used to start the worker process
+    std::vector<std::string> worker_command;
+    /// The pool of idle non-actor workers.
+    std::list<std::shared_ptr<Worker>> idle;
+    /// The pool of idle actor workers.
+    std::unordered_map<ActorID, std::shared_ptr<Worker>> idle_actor;
+    /// All workers that have registered and are still connected, including both
+    /// idle and executing.
+    // TODO(swang): Make this a map to make GetRegisteredWorker faster.
+    std::list<std::shared_ptr<Worker>> registered_workers;
+    /// All drivers that have registered and are still connected.
+    std::list<std::shared_ptr<Worker>> registered_drivers;
+  };
+
+  /// A helper function that gets the pool for the given language.
+  inline State &GetStateForLanguage(const Language &language);
```

Comment:
No, because a few callers need to modify the returned reference. For example, `RegisterWorker`:
```
  auto &state = GetStateForLanguage(worker->GetLanguage());
  state.registered_workers.push_back(std::move(worker));
```

Reply:
Maybe renaming it to `GetStateForLanguageRef()` will be better?

Reply:
Even if we want to emphasize that this function returns a reference, the name should be `GetStateRefForLanguage`? Because `GetStateForLanguageRef` sounds more like a **reference of language**, not **reference of state**.

However, it seems that usually we don't emphasize **reference** in function names. I think it's okay to follow the current convention.

(Just noted that this document is out-dated, I'll update it.)